### ISSUE TITLE
dynamic_rnn() efficiency

### ISSUE THREAD

**Issue by lelouchmatlab** [tflearn/tflearn#256]
when calling tf.nn.dynamic_rnn()/tf.nn.rnn() with sequence_length as an argument, a lot computation actually happened on CPU, because math_ops.reduce_min/max(sequence_length) can only be done on cpu? This kind of makes the RNN training slower compared with manually doing masking. Any plan to add more flexibility to support tf.rnn.run() completely on GPU (with manual mask as post-process)?

**Comment by aymericdamien** [comment]
hmm, I think this should be changed in tensorflow itself, you may post that in tensorflow issue section. Also, to speed-up RNNs computation, it would be nice to implement RNNs using tf.scan() in TFLearn, it should give great speed-up.
